src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -483,6 +483,75 @@ void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
   return_address_manager_->ProcessFunctionExit(event_data.tid);
 }
 
+[[nodiscard]] static std::shared_ptr<unwindstack::MapInfo> FindFileMapInfoPrecedingAnonMapInfo(
+    const std::shared_ptr<unwindstack::MapInfo>& anon_map_info) {
+  ORBIT_CHECK(anon_map_info->name().empty());
+  std::shared_ptr<unwindstack::MapInfo> map_info_it = anon_map_info->prev_map();
+  while (map_info_it != nullptr &&
+         (map_info_it->name().empty() || map_info_it->name().c_str()[0] == '[')) {
+    map_info_it = map_info_it->prev_map();
+  }
+
+  if (map_info_it == nullptr || (map_info_it->flags() & unwindstack::MAPS_FLAGS_DEVICE_MAP) != 0) {
+    // This is unexpected if anon_map_info was detected to belong to a PE.
+    return nullptr;
+  }
+  return map_info_it;
+}
+
+[[nodiscard]] static std::shared_ptr<unwindstack::MapInfo> FindFirstMapInfoForSameFile(
+    const std::shared_ptr<unwindstack::MapInfo>& file_map_info) {
+  const std::string file_path = file_map_info->name();
+  ORBIT_CHECK(!file_path.empty());
+  std::shared_ptr<unwindstack::MapInfo> first_map_info_for_file_path;
+  // Scan the maps backwards. Stop when a file mapping for a different file is found.
+  std::shared_ptr<unwindstack::MapInfo> map_info_it = file_map_info;
+  while (map_info_it != nullptr &&
```

Comment:
So just as a refresher for me. We assume that maps for different files don't intersect, but the "special" mappings (that starts with "[") may intersect a file mapping "range". Is that correct?

Reply:
This goes back to https://github.com/google/orbit/pull/3590#discussion_r865656667. In practice I guess you can say that we allow "special" mappings to intersect a file mapping, but the idea here is just that I'm looking for a file mapping, and I'm skipping over all anonymous mappings, be them normal or special. In other words, I only want to distinguish between file mappings and anonymous mappings, not between file/anonymous normal/anonymous special, simply because I think it's conceptually easier and it shouldn't matter in practice.

Reply:
Thanks. I was remembering that comment, but wanted to clarify. Thanks for the explanation.